tools/c7n_tencentcloud/c7n_tencentcloud/actions/core.py

```diff
@@ -0,0 +1,39 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytz
+from c7n.exceptions import PolicyValidationError
+
+from c7n.actions import BaseAction
+
+
+class TencentCloudBaseAction(BaseAction):
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super().__init__(data, manager, log_dir)
+        self.resource_type = self.manager.get_model()
+
+        self.endpoint = self.resource_type.endpoint
```

Comment:
feels like most of these instance attributes can be removed, as its just used for the get_client method which can directly reference them from the resource type. or have that method overridden for the case of tag actions.

Reply:
have updated the get_client method